### PR TITLE
Split into libudunits2 and udunits2

### PR DESCRIPTION
**Pull request by zklaus**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] ~Reset the build number to `0` (if the version changed)~
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by zklaus** [comment]
I think this is ready, @conda-forge/udunits2. I have left the artifact storage activated for now so you can inspect the packages from the artifacts as part of the review if you choose to do so. Consequently, this needs to be deactivated before the merging, after the review.

File: recipe/meta.yaml
Comment (xylar) [inline recipe/meta.yaml:40]:
```suggestion
      - lib/libudunits2.so*  # [linux]
```
It looks like `lib/libudunits2.so` is ending up in the `udunits2` package, not in `libudunits2`.

File: recipe/meta.yaml
Comment (xylar) [inline recipe/meta.yaml:73]: Jus for my education, could you explain why this is needed here but not in `libudunits2`?  I see that `expat` does things this way, too, so I trust that it's right.  I just don't really understand `run_exports` very well and would like to.

**Review by xylar — COMMENTED** [review]
A few quick comments.  I'll take a more in-depth look soon when I have a little more time.

File: recipe/meta.yaml
Comment (zklaus) [reply recipe/meta.yaml:40]: It does, but that's correct. When another package is built against Libudunits2, it will use `libudunits2.so` to find the latest version of the library. But as a requirement, it will record the concrete version of the shared object, c.f. the output of `ldd libnco.so`:
```
libudunits2.so.0 => /home/a002160/mambaforge/envs/evt/lib/././libudunits2.so.0 (0x00007fd4cb5d5000)
```

That's why `libudunits2.so` is needed only at build time, whereas all the `libudunits2.*` are needed at run time.

File: recipe/meta.yaml
Comment (zklaus) [reply recipe/meta.yaml:73]: I'll do my best ;) What `run_exports` does is saying "A package that has this package (`udunits2`) as a `build` or `host` requirement, should automatically get that package (`libudunits2`) as a `run` requirement.
That means that a package that depends on `udunits2` to build against the library will get an automatic `run` requirement that makes sure the dynamic library is available at run time.

On the other hand, it doesn't really make sense to have `libudunits2` as a `build` or `host` requirement because that's not enough to build against it.

Does that make sense?

File: recipe/meta.yaml
Comment (xylar) [inline recipe/meta.yaml:41]:
```suggestion
      - lib/libudunits2.*dylib  # [osx]
```
Similar issue here, I think.

File: recipe/meta.yaml
Comment (xylar) [reply recipe/meta.yaml:73]: Got it, that makes perfect sense.

File: recipe/meta.yaml
Comment (xylar) [reply recipe/meta.yaml:40]: Thanks for the explanation.

**Review by xylar — APPROVED** [review]
With my confusion resolved, this looks good to me. But I have also revealed my lack of expertise, so my opinion shouldn't count for too much...

**Comment by xylar** [comment]
@DPeterK, @kwilcox, @ocefpaf and @pelson, I'm not sure how active some of you are on conda-forge but it would be great to have feedback from at least one of you and then get this PR merged.

**Comment by ocefpaf** [comment]
> @DPeterK, @kwilcox, @ocefpaf and @pelson, I'm not sure how active some of you are on conda-forge but it would be great to have feedback from at least one of you and then get this PR merged.

@xylar, apologies but lately I'm reviewing and merging only the "easy stuff." Mostly b/c I'm on mobile only as b/c I'm working on my house renovations.

With that said, this PR looks good to me and I'd be keen on merging it. If we don't hear from other today I'd just merge and see how things go.

**Comment by xylar** [comment]
@zklaus if you want to update one or another of the downstream feedstocks to show how this needs to be handled, I'm happy to take care of it on any where I'm a maintainer. It seems like it might be as simple as removing `udunits2` from the `run` dependencies and relying on the `run_exports` to bring in `libudunits2` automatically if I understood correctly.

**Pull request merged**
